include/dxc/Test/HlslTestUtils.h

```diff
@@ -517,6 +542,44 @@ inline bool isnanFloat16(uint16_t val) {
 uint16_t ConvertFloat32ToFloat16(float val) throw();
 float ConvertFloat16ToFloat32(uint16_t val) throw();
 
+inline bool CompareDoubleULP(
+    const double &Src, const double &Ref, int64_t ULPTolerance,
+    hlsl::DXIL::Float32DenormMode Mode = hlsl::DXIL::Float32DenormMode::Any) {
+  if (Src == Ref) {
+    return true;
+  }
+  if (std::isnan(Src)) {
+    return std::isnan(Ref);
+  }
+
+  if (Mode == hlsl::DXIL::Float32DenormMode::Any) {
```

Comment:
Just FYI: denorms are required to be preserved for double-precision DXIL operations.  So either this should default to `Float32DenormMode::Preserve`, or we should just remove this branch.

Reply:
I can address in this PR. Conflicts is this file will be trivial in the diff for merging the rest.